src/opal.c

```diff
@@ -2213,10 +2213,22 @@ traverse_ast(node_s *node, FILE *dest_fp)
 void
 add_asm_code (asm_code_e code, int intval, char *label)
 {
-  // Create struct with given values and append to asm_cmd_list[]
-  // TBD
+  /// Create struct with given intval and code
+  asm_cmd_e asm_cmd = { 0 };
+  asm_cmd.intval = intval;
+  asm_cmd.cmd = code;
+
+  /// Add the asm_code label if there is one
+  if (label)
+  {
+      asm_cmd.label = calloc (64, sizeof(char));
```

Comment:
You don't need to malloc before `strdup`. It will give you the resultant string pointer as a return value. So you can remove this line.